xbmc/dialogs/GUIDialogSmartPlaylistEditor.cpp

```diff
@@ -183,6 +186,29 @@ bool CGUIDialogSmartPlaylistEditor::OnMessage(CGUIMessage& message)
   return CGUIDialog::OnMessage(message);
 }
 
+void CGUIDialogSmartPlaylistEditor::OnPopupMenu(int item)
+{
+  if (item < 0 || item >= m_ruleLabels->Size())
+    return;
+  if (m_playlist.m_ruleCombination.m_rules.size() == 1 && m_playlist.m_ruleCombination.m_rules[0]->m_field == FieldNone)
+    return;
+  // highlight the item
+  (*m_ruleLabels)[item]->Select(true);
+
+  CContextButtons choices;
+  choices.Add(1, 15015);
+
+  CFileItemPtr itemPtr = m_ruleLabels->Get(item);
+
+  int button = CGUIDialogContextMenu::ShowAndGetChoice(choices);
+
+  // unhighlight the item
+  (*m_ruleLabels)[item]->Select(false);
```

Comment:
`m_ruleLabels->Get(item)` is more readable. Same above. And you aren't using `itemPtr` here